jaide/cli.py

```diff
@@ -343,25 +345,45 @@ def commit(ctx, commands, blank, check, sync, comment, confirm, at_time):
                   |      seconds)
     @type at_time: str
 
+    @param action: A string set to either merge or override, to be used with stanza config_format
+    @type action: str
+    @param config_format: A string set to either set or stanza to denote the format 
+                 | of the configuration to be commited. Defaults to set.
+    @type config_format: str
+
     @returns: None. Functions part of click relating to the command group
             | 'main' do not return anything. Click handles passing context
             | between the functions and maintaing command order and chaining.
     """
     if not blank and commands == 'annotate system ""':
         raise click.BadParameter("--blank and the commands argument cannot"
                                  " both be omitted.")
+    config_format = config_format.lower()
+    action = action.lower()
+    # if using set, then format is text and action is set
+    if config_format == 'set':
+        action = 'set'
+        config_format = 'text'
+    # if using stanza, format is xml and action must be stanza or override
+    elif config_format == 'stanza':
+        if action != 'merge' and action != 'override':
+            raise click.BadParameter('When using stanza config format, action must be merge or override')
+        else:
+            config_format='xml'
+    else:
+        raise click.BadParameter("--format must be set or stanza.")
```

Comment:
the validation should move to a callback function on the option, to better separation argument validation from workflow logic.